src/github.com/travis-ci/worker/image/env_selector.go

```diff
@@ -0,0 +1,146 @@
+package image
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/travis-ci/worker/config"
+)
+
+var (
+	nonAlphaNumRegexp = regexp.MustCompile(`[^a-zA-Z0-9_]+`)
+)
+
+// EnvSelector implements Selector for environment-based mappings
+type EnvSelector struct {
+	c *config.ProviderConfig
+
+	imageAliases map[string]string
+}
+
+// NewEnvSelector builds a new EnvSelector from the given *config.ProviderConfig
+func NewEnvSelector(c *config.ProviderConfig) *EnvSelector {
+	es := &EnvSelector{c: c}
+	es.buildImageAliasMap()
+	return es
+}
+
+func (es *EnvSelector) buildImageAliasMap() {
+	aliasNames := es.c.Get("IMAGE_ALIASES")
+
+	aliasNamesSlice := strings.Split(aliasNames, ",")
+
+	imageAliases := map[string]string{}
+
+	es.c.Each(func(key, value string) {
+		if strings.HasPrefix(key, "IMAGE_") {
+			imageAliases[strings.ToLower(strings.Replace(key, "IMAGE_", "", -1))] = value
+		}
+	})
+
+	for _, aliasName := range aliasNamesSlice {
+		normalizedAliasName := strings.ToUpper(string(nonAlphaNumRegexp.ReplaceAll([]byte(aliasName), []byte("_"))))
+
+		key := fmt.Sprintf("IMAGE_ALIAS_%s", normalizedAliasName)
+		if !es.c.IsSet(key) {
+			// TODO: warn?
```

Comment:
We should probably do _something_ here, since the configuration is invalid. This happens during the worker "boot" process as far as I can tell, so might be as easy as just `panic`-ing?